### PR TITLE
fix(run/django): Update to Procfile, Artifact Registry, dedicated service account

### PR DESCRIPTION
**Pull request by glasnt**
## Description

Fixes b/364075989, cl/688789319

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist
- [ ] I have followed [Sample Guidelines from AUTHORING_GUIDE.MD](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md)
- [ ] README is updated to include [all relevant information](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#readme-file)
- [x] **Tests** pass:   `nox -s py-3.9` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [x] **Lint** pass:   `nox -s lint` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample
- [x] Please **merge** this PR for me once it is approved

### CODE REVIEW

**Comment by glasnt** [comment]
Blocked on cl/691542544 to pin region tag that is being deprecated.

**Comment by glasnt** [comment]
I can't seem to re-run snippetbot here :(

File: run/django/e2e_test_setup.yaml [hunk 1]
```diff
@@ -47,124 +47,105 @@ steps:
...
+        --project ${PROJECT_ID}"
 
   - id: "Build Container Image"
-    name: "gcr.io/cloud-builders/docker"
-    entrypoint: "/bin/bash"
-    args:
-      - "-c"
-      - |
-        ./retry.sh "docker build -t gcr.io/${PROJECT_ID}/${_SERVICE}:${_VERSION} ."
+    name: gcr.io/k8s-skaffold/pack
```
Comment (iennae) [inline run/django/e2e_test_setup.yaml:72]: this look like it's deprecated/unsupported

Comment (glasnt) [reply run/django/e2e_test_setup.yaml:72]: It's the one currently listed on https://cloud.google.com/docs/buildpacks/build-application#build_an_application_remotely

If that changes, we should promptly update all samples that uses the old image.

Comment (glasnt) [reply run/django/e2e_test_setup.yaml:72]: I can try the (probable) replacement "buildpacksio/pack", as this seems to work in isolation.

**Review by iennae — APPROVED** [review]
This looks good to me. Thanks for the thorough update.

**Pull request merged**